src/WinGetServer/WinMain.cpp

```diff
@@ -18,27 +27,139 @@ static void _releaseNotifier() noexcept
+
+    UINT32 bufferSize = static_cast<UINT32>(streamSize.QuadPart);
+
+    struct DeleteWithMidlFree { void operator()(void* m) { MIDL_user_free(m); } };
+    std::unique_ptr<BYTE, DeleteWithMidlFree> buffer{ reinterpret_cast<BYTE*>(MIDL_user_allocate(bufferSize)) };
+
+    RETURN_IF_FAILED(stream->Seek({}, STREAM_SEEK_SET, nullptr));
+    ULONG bytesRead = 0;
+    RETURN_IF_FAILED(stream->Read(buffer.get(), bufferSize, &bytesRead));
+    RETURN_HR_IF(E_UNEXPECTED, bytesRead != bufferSize);
+
+    *pcbBuffer = bufferSize;
+    *ppBuffer = buffer.release();
+
+    return S_OK;
+}
+
+int __stdcall wWinMain(_In_ HINSTANCE, _In_opt_ HINSTANCE, _In_ LPWSTR cmdLine, _In_ int)
+{
+    wil::SetResultLoggingCallback(&WindowsPackageManagerServerWilResultCallback);
+
+    RETURN_IF_FAILED(CoInitializeEx(nullptr, COINIT_MULTITHREADED));
 
     // Enable fast rundown of objects so that the server exits faster when clients go away.
     {
-        winrt::com_ptr<IGlobalOptions> globalOptions;
-        winrt::check_hresult(CoCreateInstance(CLSID_GlobalOptions, nullptr, CLSCTX_INPROC, IID_PPV_ARGS(&globalOptions)));
-        winrt::check_hresult(globalOptions->Set(COMGLB_RO_SETTINGS, COMGLB_FAST_RUNDOWN));
+        wil::com_ptr<IGlobalOptions> globalOptions;
+        RETURN_IF_FAILED(CoCreateInstance(CLSID_GlobalOptions, nullptr, CLSCTX_INPROC, IID_PPV_ARGS(&globalOptions)));
+        RETURN_IF_FAILED(globalOptions->Set(COMGLB_RO_SETTINGS, COMGLB_FAST_RUNDOWN));
     }
 
     RETURN_IF_FAILED(WindowsPackageManagerServerInitialize());
 
+    // Command line parsing
+    int argc = 0;
+    LPWSTR* argv = CommandLineToArgvW(cmdLine, &argc);
+    RETURN_LAST_ERROR_IF(!argv);
+
+    bool manualActivation = false;
+
+    // If command line gets more complicated, consider more complex parsing
+    if (argc == 1 && std::wstring_view{ L"--manualActivation" } == argv[0])
+    {
+        manualActivation = true;
+    }
+
     _comServerExitEvent.create();
     RETURN_IF_FAILED(WindowsPackageManagerServerModuleCreate(&_releaseNotifier));
     try
     {
         // Register all the CoCreatableClassWrlCreatorMapInclude classes
         RETURN_IF_FAILED(WindowsPackageManagerServerModuleRegister());
+
+        if (manualActivation)
+        {   
+            HANDLE hMutex = NULL;
+            hMutex = CreateMutex(NULL, FALSE, TEXT("WinGetServerMutex"));
+            RETURN_HR_IF_NULL(OSS_MUTEX_NOT_CREATED, hMutex);
+
+            DWORD waitResult = WaitForSingleObject(hMutex, 0);
+            if (waitResult != 0 && waitResult != WAIT_ABANDONED)
+            {
+                return ERROR_SERVICE_ALREADY_RUNNING;
+            }
+
+            RETURN_IF_FAILED(WindowsPackageManagerServerInitializeRPCServer());
+        }
+
+        // Manual reset event to notify the client that the server is available.
+        HANDLE eventHandle = CreateEventW(NULL, TRUE, FALSE, L"WinGetServerStartEvent");

```

Comment:
>HANDLE eventHandle = CreateEventW(NULL, TRUE, FALSE, L"WinGetServerStartEvent"); [](http://example.com/codeflow?start=8&length=80)

Please use wil::unique_event for dealing with events if possible, the wil implementation has easy to use wraps for create(), set(), wait()

Reply:
changed to using wil::unique_event for dealing with the manual reset event.